src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -114,144 +257,1279 @@ static DamageTable D_8089E314 = {
+            func_800B9010(&this->actor, 0x20B2);
+        }
+    }
+}
+
+s32 func_8089DC4C(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
+                  Gfx** gfx) {
+    EnDinofos* this = THIS;
+
+    if (limbIndex == 16) {
+        rot->y -= this->unk_28E;
+    }
+
+    return 0;
+}
+
+void func_8089DC84(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
+    static Vec3f D_8089E38C = { 400.0f, -3600.0f, 0.0f };
+    static Vec3f D_8089E398 = { 300.0f, 500.0f, 0.0f };
+    static Vec3f D_8089E3A4 = { 700.0f, 400.0f, 0.0f };
+    static s8 D_8089E3B0[] = {
+        255, 255, 0, 255, 1, 2, 255, 3, 4, 255, 5, 6, 255, 7, 8, 255, 9, 10, 255, 11, 255,
+    };
+
+    EnDinofos* this = THIS;
+    Vec3f sp80;
+    Vec3f sp74;
+    Vec3f sp68;
+    Vec3f sp5C;
+    s32 sp58;
+    Vec3f sp4C;
+    MtxF* sp48;
+
+    Collider_UpdateSpheres(limbIndex, &this->colliderJntSph);
+    if (D_8089E3B0[limbIndex] != -1) {
+        SysMatrix_GetStateTranslation(&this->unk_2D4[D_8089E3B0[limbIndex]]);
+    }
+
+    if ((limbIndex == 14) && (this->unk_292 != this->unk_290) &&
+        ((this->actionFunc == func_8089C56C) || (this->actionFunc == func_8089C2A8))) {
+        Math_Vec3f_Copy(&sp74, &this->colliderQuad.dim.quad[0]);
+        Math_Vec3f_Copy(&sp80, &this->colliderQuad.dim.quad[1]);
+        SysMatrix_MultiplyVector3fByState(&D_8089E38C, &sp68);
+        SysMatrix_MultiplyVector3fByState(&D_8089E398, &sp5C);
+        Collider_SetQuadVertices(&this->colliderQuad, &sp5C, &sp68, &sp74, &sp80);
+        if (this->colliderQuad.base.atFlags & 1) {
+            func_800A81F0(Effect_GetParams(this->unk_2A0), &sp68, &sp5C);
+        }
+        this->unk_292 = this->unk_290;
+    }
+
+    if (limbIndex == 4) {
+        SysMatrix_GetStateTranslationAndScaledX(300.0f, &this->actor.shape.feetPos[0]);
+    } else if (limbIndex == 7) {
+        SysMatrix_GetStateTranslationAndScaledX(300.0f, &this->actor.shape.feetPos[1]);
+    }
+
+    if ((limbIndex == 16) && (this->actionFunc == func_8089B580)) {
+        if ((this->skelAnime.animCurrentFrame > 38.0f) && (this->skelAnime.animCurrentFrame <= 55.0f) &&
+            (this->unk_292 != this->unk_290)) {
+            sp48 = SysMatrix_GetCurrentState();
+            sp58 = 48 - (s32)this->skelAnime.animCurrentFrame;
+            if (sp58 < 0) {
+                sp58 = 0;
+            }
+            SysMatrix_MultiplyVector3fByState(&D_8089E3A4, &sp4C);
+            sp4C.x -= sp48->mf[3][0];
+            sp4C.y -= sp48->mf[3][1];
+            sp4C.z -= sp48->mf[3][2];
+            EffectSsDFire_Spawn(globalCtx, &this->unk_34C, &sp4C, &D_801D15B0, 30, 22, 255 - (sp58 * 20), 20, 3, 8);
+            this->unk_292 = this->unk_290;
+        }
+    }
+}
+
+void EnDinofos_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnDinofos* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if (this->unk_288 == 0xFF) {
```

Comment:
```suggestion
    if (this->unk_288 == 255) {
```
This is likely an alpha value if I had to guess